### PR TITLE
address_module

### PR DESCRIPTION
**Pull request by nikpawar89**
changes suggested have been implemented

### CODE REVIEW

File: app/scripts/controllers/client/CreateClientController.js [hunk 1]
```diff
@@ -51,8 +62,62 @@
...
+
+                scope.addressTypes=data.address.addressTypeIdOptions;
+                scope.countryOptions=data.address.countryIdOptions;
+                scope.stateOptions=data.address.stateProvinceIdOptions;
+                scope.enableAddress=data.isAddressEnabled;
+
+                if(scope.enableAddress===true)
+                {
+                    resourceFactory.addressFieldConfiguration.get({entity:entityname},function(data){
+                        scope.addresstypId=data[0].is_enabled;
```
Comment (nazeer1100126) [inline app/scripts/controllers/client/CreateClientController.js]: Usually, we should not have hard coded indexes like this.  Problem with this implementation is in future somebody adds a new type in between these indexes, that person need to comeback and look into these existing indexes also.

File: app/scripts/controllers/client/CreateClientController.js [hunk 2]
```diff
@@ -127,6 +192,74 @@
                     this.formData.savingsProductId = null;
                 }
 
+                if(scope.enableAddress===true)
+                {
+                    for(var i=0;i<scope.addressArray.length;i++)
+                    {
+                        var temp=new Object();
+                        if(scope.addressArray[i].addressTypeId)
```
Comment (nazeer1100126) [inline app/scripts/controllers/client/CreateClientController.js]: If address type id is not enabled and rest of them are enabled in which address category you are going to save this new address?

File: app/scripts/controllers/client/ViewClientController.js [hunk 3]
```diff
@@ -12,6 +12,85 @@
...
+
+
+            resourceFactory.clientTemplateResource.get(function(data)
+            {
+                scope.enableAddress=data.isAddressEnabled;
+                if(scope.enableAddress===true)
+                {
+
+                    resourceFactory.addressFieldConfiguration.get({entity:entityname},function(data){
+                        scope.view.addresstypId=data[0].is_enabled;
```
Comment (nazeer1100126) [inline app/scripts/controllers/client/ViewClientController.js]: Usually, we should not have hard coded indexes like this. Problem with this implementation is in future somebody adds a new type in between these indexes, that person need to comeback and look into these existing indexes also

File: app/scripts/controllers/configurations/AddressFormController.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            resourceFactory.clientaddressFields.get(function(data){
+                $scope.addressTypes=data.addressTypeIdOptions;
+                $scope.countryOptions=data.countryIdOptions;
+                $scope.stateOptions=data.stateProvinceIdOptions;
+            }
+            )
+
+
+            resourceFactory.addressFieldConfiguration.get({entity:entityname},function(data){
+                $scope.addresstypId=data[0].is_enabled;
```
Comment (nazeer1100126) [inline app/scripts/controllers/configurations/AddressFormController.js]: Usually, we should not have hard coded indexes like this. Problem with this implementation is in future somebody adds a new type in between these indexes, that person need to comeback and look into these existing indexes also

File: app/scripts/controllers/client/ViewClientController.js [hunk 5]
```diff
@@ -12,6 +12,85 @@
...
+                        scope.view.longitude=data[12].is_enabled;
+                        scope.view.isActive=data[17].is_enabled;
+                    })
+
+                    // resourceFactory.clientAddress.get({clientId:scope.client.id},function(data)routeParams.id
+                    resourceFactory.clientAddress.get({clientId:routeParams.id},function(data)
+                    {
+
+                        scope.addresses=data;
+                        if(data[0])
```
Comment (nazeer1100126) [inline app/scripts/controllers/client/ViewClientController.js]: Don't use hard coded indexes. Your check should have like data.addressTypeId

File: app/scripts/controllers/client/ViewClientController.js [hunk 6]
```diff
@@ -12,6 +12,85 @@
...
+                        scope.view.state_province=data[8].is_enabled;
+                        scope.view.country=data[9].is_enabled;
+                        scope.view.postal_code=data[10].is_enabled;
+                        scope.view.latitue=data[11].is_enabled;
+                        scope.view.longitude=data[12].is_enabled;
+                        scope.view.isActive=data[17].is_enabled;
+                    })
+
+                    // resourceFactory.clientAddress.get({clientId:scope.client.id},function(data)routeParams.id
+                    resourceFactory.clientAddress.get({clientId:routeParams.id},function(data)
```
Comment (nazeer1100126) [inline app/scripts/controllers/client/ViewClientController.js]: Is it possible to have multiple address for a client? If yes, where you are handling it?

File: app/scripts/routes.js [hunk 7]
```diff
@@ -7,6 +7,9 @@
             .when('/login', {
                 templateUrl: 'views/login.html'
             })
+            .when('/login', {
```
Comment (nazeer1100126) [inline app/scripts/routes.js]: why do we need to have two /login entries? By mistake added?

File: app/scripts/controllers/configurations/EditAddressController.js [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+            resourceFactory.clientaddressFields.get(function(data){
+                    $scope.addressTypes=data.addressTypeIdOptions;
+                    $scope.countryOptions=data.countryIdOptions;
+                    $scope.stateOptions=data.stateProvinceIdOptions;
+                }
+            )
+            // clientId=sharedVariables.getClientId();
+
+            resourceFactory.addressFieldConfiguration.get({entity:entityname},function(data){
+                $scope.addresstypId=data[0].is_enabled;
```
Comment (nazeer1100126) [inline app/scripts/controllers/configurations/EditAddressController.js]: Same comment as above. Not suggested to use hard coded indexes

File: app/scripts/controllers/configurations/AddressFormController.js (hunk 4 above)
Comment (nikpawar89) [inline app/scripts/controllers/configurations/AddressFormController.js]: Address type would be made mandatory field

File: app/scripts/routes.js (hunk 7 above)
Comment (nikpawar89) [inline app/scripts/routes.js]: This is an error. Would clean it up

File: app/scripts/controllers/client/CreateClientController.js (hunk 2 above)
Comment (nikpawar89) [inline app/scripts/controllers/client/CreateClientController.js]: Address type Id would made mandatory. There is a column in the m_field_configuration table which marks the filed as mandatory.

**Pull request closed without merging**